crypto/evp/e_aes.c

```diff
@@ -3510,6 +3510,7 @@ static int aes_xts_init_key(EVP_CIPHER_CTX *ctx, const unsigned char *key,
 static int aes_xts_cipher(EVP_CIPHER_CTX *ctx, unsigned char *out,
                           const unsigned char *in, size_t len)
 {
+    const size_t blocks = len / AES_BLOCK_SIZE + (len % AES_BLOCK_SIZE != 0);
```

Comment:
There is a nicer way of doing this

Reply:
The usual `(len + AES_BLOCK_SIZE - 1) / AES_BLOCK_SIZE` risks overflow here.  Hence the safe option.

Reply:
Why not avoid both the potential overflow and the double division (`/` and `%`) by comparing the length instead of the block count below?

```C
    if (len > (1<<20) * AES_BLOCK_SIZE)
        return 0;
```
Since the right hand side is a constant expression, it can't overflow.

Also, even though there is a comment explaining the constant `(1<<20)` I still would prefer if all those limits imposed by NIST would be properly defined as macros with some consistent naming (starting with a common prefix like `NIST_...`) instead of using magic numbers.  Something like `NIST_AES_XTS_MAX_BLOCK_SIZE` or whatever you prefer.

Reply:
Maybe it would even make sense to collect these limits in a common header file?

Reply:
Common header made.  Although, it only has one entry at the moment.